icu4c/source/i18n/unum.cpp

```diff
@@ -177,7 +177,9 @@ unum_clone(const UNumberFormat *fmt,
     } else {
         const RuleBasedNumberFormat* rbnf = dynamic_cast<const RuleBasedNumberFormat*>(nf);
         U_ASSERT(rbnf != nullptr);
-        res = rbnf->clone();
+        if (rbnf != nullptr) {
+            res = rbnf->clone();
+        }
```

Comment:
This does not look right. If the input was a nullptr, then the following code would misleadingly report a memory-allocation-error.

I am fine with guarding against fmt==nullptr although it's a little weird to call a cloning function with a nullptr.
We would need to look at other C API uprefix_clone() functions for consistency:
We could detect fmt==nullptr early, right after the U_FAILURE check, and either just return nullptr, or maybe set a U_ILLEGAL_ARGUMENT_ERROR.

Reply:
> We would need to look at other C API uprefix_clone() functions for consistency:
> We could detect fmt==nullptr early, right after the U_FAILURE check, and either just return nullptr, or maybe set a U_ILLEGAL_ARGUMENT_ERROR.

I don't have a spare time to rework a patch this way, sorry.